apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php

```diff
@@ -15,51 +15,58 @@ function __construct() {
 		}
 		self::$initialized = true;
 
-		add_action( 'init', array( $this, 'register_script_and_style' ), 100 );
 		add_action( 'init', array( $this, 'register_blocks' ), 100 );
 		add_action( 'init', array( $this, 'register_wp_template' ) );
+
+		add_action( 'enqueue_block_editor_assets', array( $this, 'enqueue_script_and_style' ), 100 );
 	}
 
 	function register_wp_template() {
 		require_once plugin_dir_path( __FILE__ ) . 'wp-template.php';
 		fse_register_wp_template();
 	}
 
-	function register_script_and_style() {
+	function enqueue_script_and_style() {
+		if ( ! $this->is_editor_wp_template() ) {
+			return;
+		}
+
 		$script_dependencies = json_decode( file_get_contents(
 			plugin_dir_path( __FILE__ ) . 'dist/full-site-editing-plugin.deps.json'
 		), true );
-		wp_register_script(
+		wp_enqueue_script(
 			'a8c-full-site-editing-script',
 			plugins_url( 'dist/full-site-editing-plugin.js', __FILE__ ),
 			is_array( $script_dependencies ) ? $script_dependencies : array(),
-			filemtime( plugin_dir_path( __FILE__ ) . 'dist/full-site-editing-plugin.js' )
+			filemtime( plugin_dir_path( __FILE__ ) . 'dist/full-site-editing-plugin.js' ),
+			true
 		);
 
 		$style_file = is_rtl()
 			? 'full-site-editing-plugin.rtl.css'
 			: 'full-site-editing-plugin.css';
-		wp_register_style(
+		wp_enqueue_style(
 			'a8c-full-site-editing-style',
 			plugins_url( 'dist/' . $style_file, __FILE__ ),
 			array(),
 			filemtime( plugin_dir_path( __FILE__ ) . 'dist/' . $style_file )
 		);
 	}
 
-	// We only need to declare script and style as dependencies once
-	// Because they'll be then enqueued for every block.
 	function register_blocks() {
 		register_block_type( 'a8c/content-slot', array(
-			'editor_script'   => 'a8c-full-site-editing-script',
-			'editor_style'    => 'a8c-full-site-editing-style',
 			'render_callback' => 'render_content_slot_block',
 		 ) );
 
 		register_block_type( 'a8c/template-part', array(
 			'render_callback' => 'render_template_part_block',
 		) );
 	}
+
+	function is_editor_wp_template() {
+		$current_screen = get_current_screen();
+		return 'wp_template' === $current_screen->post_type;
```

Comment:
Since `$current_screen` isn't being used for anything else I'd probably just rewrite this as a one-liner:

```php
return 'wp_template' === get_current_screen()->post_type;
```